modules/media_duplicate_validation/media_duplicate_validation.module

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * @file
+ * Contains media_duplicate_validation.module.
+ */
+
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_help().
+ */
+function media_duplicate_validation_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the media_duplicate_validation module.
+    case 'help.page.media_duplicate_validation':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Media Validation plugins to help prevent duplication of media items') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements hook_cache_flush().
+ */
+function media_duplicate_validation_cache_flush() {
+  /** @var \Drupal\media_duplicate_validation\Plugin\MediaDuplicateValidationManager $duplicate_manager */
+  $duplicate_manager = \Drupal::service('plugin.manager.media_duplicate_validation');
+  $duplicate_manager->buildPluginSchemas();
```

Comment:
Why are you rebuilding the schema on every cache flush?

Reply:
Plugins are recognized during cache flush. within the manager, it only creates the schema if the table doesnt exist

Reply:
That leads to the question, why would the table need to be created on cache flush? Why would it not exist? Is this to prevent some other bad thing that is happening?

Reply:
a plugin doesn't natively create/use tables. so we have to do so after the plugin is discovered which is on cache flush.. and on module enable i suppose.

Reply:
Module enable/update is the appropriate place for this action.

Reply:
I'm moving it to a module_installed hook and adding a module_preuninstall hook to clean up schemas

Reply:
Digging in to this more, the buildPluginSchemas function also loads up all media and adds it to a cron queue to be processed and inserted in to the DB table. I'd like that to only happen once.

Reply:
it only loads all media entities into the queue if the table doesnt exist. so it will only do that one time for each plugin unless for some reason that table is dropped. I also changed is so that it does a db query and gets the mids instead of loading all entities. reducing the memory load and execution time

Reply:
`hook_module_installed` would be neccessary if a 3rd party module ever gets enabled that contains a plugin. even if it gets called twice, it wouldn't double up the cron queue.

Reply:
`hook_install` is called before `hook_module_installed` so really theres currently no need for the hook_install(). i'll remove